app/main.py

```diff
@@ -1,2 +1,98 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.pairs = [[] for i in range(8)]
+        self.next_index = 0
+
+    def __setitem__(
+            self,
+            key: int | float | str | type | bool | tuple | frozenset,
```

Comment:
Instead of including all values in the type annotation, indicate that the key should be `Hashable`. You could import `Hashable` from `typing`. Change this annotation in all places.